Mathlib/CategoryTheory/Sites/LocallyBijective.lean

```diff
@@ -0,0 +1,152 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Sites.LocallySurjective
+import Mathlib.CategoryTheory.Sites.Localization
+
+/-!
+# Locally bijective morphisms of presheaves
+
+Let `C` a be category equipped with a Grothendieck topology `J`.
+Let `A` be a concrete category.
+In this file, we introduce a type-class `J.WEqualsLocallyBijective A` which says
+that the class `J.W` (of morphisms of presheaves which become isomorphisms
+after sheafification) is the class of morphisms that are both locally injective
+and locally surjective (i.e. locally bijective). We prove that this holds iff
+for any presheaf `P : Cᵒᵖ ⥤ A`, the sheafification map `toSheafify J P` is locally bijective.
+We show that this holds under certain universe assumptions.
+
+-/
+
+universe w' w v' v u' u
+namespace CategoryTheory
+
+variable {C : Type u} [Category.{v} C] {J : GrothendieckTopology C}
+  {A : Type u'} [Category.{v'} A] [ConcreteCategory.{w} A]
+
+namespace Sheaf
+
+section
+
+variable {F G : Sheaf J (Type w)} (f : F ⟶ G)
+
+/-- A morphism of sheaves of types is locally bijective iff it is an isomorphism.
+(This is generalized below as `isLocallyBijective_iff_isIso`.) -/
+private lemma isLocallyBijective_iff_isIso' :
+    IsLocallyInjective f ∧ IsLocallySurjective f ↔ IsIso f := by
```

Comment:
Doesn't this follow from `IsLocallyInjective f ↔ mono f` and `IsLocallyInjective f ↔ epi f` and that `Sheaf J (Type w)` is balanced?

Reply:
This lemma does not use `HasSheafify J (Type w)`, but `IsLocallySurjective f ↔ epi f` does, so that I would prefer keeping the statement of this lemma as is.